sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIO.java

```diff
@@ -1543,7 +1543,6 @@ public PDone expand(PCollection<T> input) {
                       new PreparePubsubWriteDoFn<>(
                           getFormatFn(),
                           topicFunction,
-                          getNeedsOrderingKey(),
```

Comment:
NeedsOrderingKey sounds like it if it is true, that messages without an ordering key should not be publishable.  Woudl something like usesOrderingKey or publishWithOrderingKey be clearer?

Reply:
Agreed that the wording could be better, but this option already exists with the same name for Pubsub read. This follows a pattern with other Pubsub message components as well: 
https://github.com/apache/beam/blob/12974595cd0405942ab8b90ba6475c7eae0d4769/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIO.java#L853-L857